Define the mission and vision of OpenFeature (#55)
open-feature/community: mission-vision.md

# OpenFeature Mission and Vision

## Mission

> To improve the software development lifecycle, no matter the size of the project, by standardizing feature flagging for developers.

Reviewer: The statement `To improve the software development lifecycle`, is too general. 

What about something like

OpenFeature's Mission: **remove feature flag adoption barriers by standardizing feature flagging for developers.**

Author: I hear what you're saying but I personally like that it's a bit ambitious. I think standardized feature flagging can lead to improvements across all aspects of the SDLC.

## Vision

> A world where feature flagging is a core principle of the software development lifecycle, enabling developers to test and release high quality software with a high degree of safety and confidence.

Reviewer 2: I think we could broaden it and say "software delivery teams" or just "teams".

> A world where feature flagging is a core principle of the software development lifecycle, enabling developers to test and release high quality software with a high degree of safety and confidence.

Reviewer 3: > I think we could broaden it and say "software delivery teams" or just "teams".

Oh, I like "teams".

OpenFeature can achieve this vision by:

### Vendor neutrality

OpenFeature aims to be the primary mechanism an [application author][glossary-app-auth] interacts with when using feature flags in code by standardizing the developer facing interfaces in a vendor neutral way. This provides flexibility for the [application integrators][glossary-app-int] to choose the tooling that best fits their current requirements, while avoiding code-level lock-in.

### Extensibility

> OpenFeature should be extensible while still being easy to work with. It should be quick to get started with, while providing [application authors][glossary-app-auth] the flexibility to customize the functionality based on their business requirements and workflows.

Reviewer 3: > OpenFeature should be extensible...

We say "OpenFeature is" elsewhere, and "OpenFeature should" here.

I think there's a tenancy to want to use future tense here because were just starting, but that wont be the case forever. We should perhaps say something like "OpenFeature values/prioritizes extensibility", and keep things present tense.

Author: I should use future tense everywhere since this is a subset of the vision.

Reviewer 3: I think consistency is important... I feel like present tense is the best choice, but maybe I'm wrong about that.

Author: I ended up using present tense. Please let me know if you notice any inconsistencies.

### Community

> OpenFeature is a collective effort that benefits from years of experience across multiple verticals, including many of the top feature flag vendors and practitioners. The common understanding of feature flags and their use in software, developed by this collaboration, will enable feature flags to become a core principle of the software development lifecycle.

Reviewer 3: A bit of a bigger change to the last sentence, that might be clearer and read better.

```suggestion
OpenFeature is a collective effort that benefits from years of experience across multiple verticals, including many of the top feature flag vendors and practitioners. The patterns, idioms, and techniques developed by this collaboration will enable feature flagging to become a core principle of the software development lifecycle.
```

> OpenFeature is a collective effort that benefits from years of experience across multiple verticals, including many of the top feature flag vendors and practitioners. The common understanding of feature flags and their use in software, developed by this collaboration, will enable feature flags to become a core principle of the software development lifecycle.

Reviewer 3: Both together:

```suggestion
OpenFeature is a collective effort that benefits from years of experience across multiple verticals, including top feature flag vendors and practitioners. The patterns, idioms, and techniques developed by this collaboration will enable feature flagging to become a core principle of the software development lifecycle.
```

### Education

> Feature flagging is a simple, yet powerful mechanism that improves the entire software development lifecycle by decoupling feature release from a deployment. It isn't, however, ubiquitously used throughout the industry. OpenFeature is an opportunity to promote good software practices in a vendor neutral way through feature flag evangelism.

Reviewer 3: `evangelism`: I kinda dislike this word, and I feel like the industry is moving away from it because of it's baggage.. Can we say "promoting feature flag concepts" or something like that?
